modules/@apostrophecms/login/index.js

```diff
@@ -981,6 +999,87 @@ module.exports = {
             required: true
           })
           );
+      },
+
+      normalizeLoginName(usernameOrEmail) {
+        if (typeof usernameOrEmail !== 'string' || !self.options.caseInsensitive) {
+          return usernameOrEmail;
+        }
+        return usernameOrEmail.toLowerCase();
+      },
+
+      async manageLoginCaseInsensitiveMigration() {
+        if (self.options.caseInsensitive) {
+          self.apos.migration.add('login-case-insensitive', async () => {
+            await self.caseInsensitiveTask();
+          });
+        } else {
+          self.apos.migration.cancel('login-case-insensitive');
+        }
```

Comment:
If caseInsensitive is activated then deactivated. We remove the migration to re run it if the option is re activated. Makes sense to me to run it anytime you switch from sensitive to insensitive..
WDYT?

Reply:
This will add it yes and run it, one time ever, as a migration. For weird edge cases like turning it on and off again and on again, they should use the task.